orders/models.py

```diff
@@ -177,25 +172,41 @@ def build_order_rows(self):
 
         rows = []
 
-        days_for_self = self.unconfirmed_details['days_for_self']
-        if days_for_self is not None:
-            ticket = Ticket.objects.build(
-                rate=self.unconfirmed_details['rate'],
-                owner=self.purchaser,
-                days=days_for_self,
-            )
-            row = self.order_rows.build_for_item(ticket)
-            rows.append(row)
+        if self.content_type == ContentType.objects.get(app_label="tickets", model="ticket"):
 
-        email_addrs_and_days_for_others = self.unconfirmed_details['email_addrs_and_days_for_others']
-        if email_addrs_and_days_for_others is not None:
-            for email_addr, name, days in email_addrs_and_days_for_others:
+            days_for_self = self.unconfirmed_details['days_for_self']
+            if days_for_self is not None:
                 ticket = Ticket.objects.build(
                     rate=self.unconfirmed_details['rate'],
-                    email_addr=email_addr,
-                    days=days,
+                    owner=self.purchaser,
+                    days=days_for_self,
                 )
-                rows.append(self.order_rows.build_for_item(ticket))
+                row = self.order_rows.build_for_item(ticket)
+                rows.append(row)
+
+            email_addrs_and_days_for_others = self.unconfirmed_details['email_addrs_and_days_for_others']
+            if email_addrs_and_days_for_others is not None:
+                for email_addr, name, days in email_addrs_and_days_for_others:
+                    ticket = Ticket.objects.build(
+                        rate=self.unconfirmed_details['rate'],
+                        email_addr=email_addr,
+                        days=days,
+                    )
+                    rows.append(self.order_rows.build_for_item(ticket))
+
+        elif self.content_type == ContentType.objects.get(app_label="extras", model="childrenticket"):
+
+            children_ticket_content_type = ContentType.objects.get(
+                app_label="extras", model="childrenticket"
+            )
+
+            ticket = ExtraItem.objects.build(
+                content_type=children_ticket_content_type,
+                owner=self.purchaser,
+                details=self.unconfirmed_details,
+            )
+            row = self.order_rows.build_for_item(ticket)
+            rows.append(row)
```

Comment:
Add an `else: assert False` here?

Reply:
You know how I feel about asserts in code 😄